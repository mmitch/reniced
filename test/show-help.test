#!/bin/bash
set -e -o pipefail

# get current version number
VERSION=$(grep 'reniced [0-9]' ../reniced | head -n 1 | sed -e 's/^\s*reniced\s*//' -e 's/\s*$$//')

# replace version number with placeholder in output
../reniced -h | sed "s/$VERSION/%%%VERSION%%%/"
