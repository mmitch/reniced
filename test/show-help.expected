Usage:
   reniced [-h] [-v] [-o format] [configfile]

Options:
     -h           print help
     -n           only show what would be done
     -v           be verbose
     -o format    output format to filter on
                  see ps(1) manpage for details
                  default: comm
     configfile   read alternative configuration file
                  default: /etc/reniced.conf for root
                           ~/.reniced for others

Configuration file format:
   # is a comment
   command perl_regular_expression

Command format:
   5      set nice value to 5
   b2     set io priority to best effort class 2
   i      set io priority to idle
   -10r4  set nice value to -10 and io priority to realtime class 4
   o-10   set OOM value /proc/$PID/oom_adj to -10

Version:
   reniced %%%VERSION%%%
